TEST_APPS/device/nfcapp/SmartPoster.cpp

```diff
@@ -0,0 +1,247 @@
+/* Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "SmartPoster.h"
+
+#include "nfc/ndef/common/Text.h"
+#include "nfc/ndef/common/URI.h"
+#include "nfc/ndef/common/Mime.h"
+#include "nfc/ndef/MessageBuilder.h"
+#include "nfc/ndef/common/util.h"
+
+using mbed::Span;
+
+using mbed::nfc::ndef::MessageBuilder;
+using mbed::nfc::ndef::RecordType;
+using mbed::nfc::ndef::Record;
+using mbed::nfc::ndef::RecordID;
+using mbed::nfc::ndef::RecordPayload;
+using mbed::nfc::ndef::common::span_from_cstr;
+using mbed::nfc::ndef::common::Mime;
+using mbed::nfc::ndef::common::Text;
+using mbed::nfc::ndef::common::URI;
+
+// todo: this class probably needs to be in the nfc module itself
```

Comment:
Please remove this todo and create internal ticket for this rather (create a new PR removing it)

Reply:
This class was lifted out of our NFC example, hence the move requirement. Thanks for reminder, jira created to move and remove.